test/solver_tests/percolation_test.py

```diff
@@ -0,0 +1,200 @@
+        # of nodes is uncertain
+        expected_edges = [
+            (7, 8),
+            (7, 10),
+            (8, "c1"),
+            (8, "c2"),
+            (8, "c3"),
+            (10, 11),
+            (10, 12),
+            (11, "c7"),
+            (11, "c6"),
+            (12, "c4"),
+            (12, "c5"),
+        ]
+        for i in expected_edges:
+            self.assertIn(i, p_tree.get_tree_topology().edges)
+
+    def test_Greedy(self):
+        cm = pd.DataFrame.from_dict(
+            {
+                "c1": [0, 4, -1, 2, 1],
+                "c2": [-1, 2, 0, 0, 1],
+                "c3": [0, 0, 0, 0, 1],
+                "c4": [5, 3, 1, 0, 0],
+                "c5": [5, 3, 1, 0, 0],
+                "c6": [0, 3, 0, 2, 4],
+                "c7": [5, 0, 1, 2, 4],
+            },
+            orient="index",
+            columns=["a", "b", "c", "d", "e"],
+        )
+        p_tree = cas.data.CassiopeiaTree(
+            cm, missing_state_indicator=-1
+        )
+
+        joining_solver = VanillaGreedySolver()
+        psolver = PercolationSolver(joining_solver = joining_solver)
+        psolver.solve(p_tree)
+        # Due to the way that networkx finds connected components, the ordering
+        # of nodes is uncertain
+        expected_edges = [
+            (7, 8),
+            (7, 10),
+            (7, "c6"),
+            (8, "c1"),
+            (8, "c2"),
+            (8, "c3"),
+            (10, "c7"),
+            (10, 11),
+            (11, "c4"),
+            (11, "c5"),
+        ]
+        for i in expected_edges:
+            self.assertIn(i, p_tree.get_tree_topology().edges)
+
+    def test_priors_case(self):
+        cm = pd.DataFrame.from_dict(
+            {
+                "c1": [0, 2, 0, -1, 1],
+                "c2": [-1, 2, 3, 0, 1],
+                "c3": [0, 2, 0, 0, 1],
+                "c4": [5, 0, 3, 0, 1],
+                "c5": [5, 3, 3, -1, 0],
+                "c6": [5, 3, 3, 2, 0],
+            },
+            orient="index",
+            columns=["a", "b", "c", "d", "e"],
+        )
+
+        priors = {
+            0: {5: 0.5},
+            1: {2: 0.6, 3: 0.4},
+            2: {3: 0.1},
+            3: {2: 0.5},
+            4: {1: 0.6},
+        }
+
+        p_tree = cas.data.CassiopeiaTree(
+            cm, missing_state_indicator=-1, priors=priors
+        )
+        joining_solver = NeighborJoiningSolver(dissimilarity_function = dissimilarity_functions.weighted_hamming_distance, add_root = True)
+        psolver = PercolationSolver(joining_solver = joining_solver)
+        psolver.solve(p_tree)
+        # Due to the way that networkx finds connected components, the ordering
+        # of nodes is uncertain
+        expected_newick_strings = [
+            "((c2,c4,(c5,c6)),(c1,c3));",
+            "((c2,c4,(c6,c5)),(c1,c3));",
+            "((c2,(c5,c6),c4),(c1,c3));",
+            "((c2,(c6,c5),c4),(c1,c3));",
```

Comment:
If you set a seed, can you make this deterministic? Also - same comment as above, I think we may want to stay away from testing newick string equality.